sdk/azidentity/live_test.go

```diff
@@ -62,7 +64,9 @@ const (
 	fakeUsername   = "fake@user"
 )
 
-var liveTestScope = "https://management.core.windows.net//.default"
+var liveTestScope = "https://management.core.windows.net//.default" // For ADFS tests, had to change this scope to openid
```

Comment:
just a small note that this liveTestScope had to change for the ADFS tests. Might have to get this value from an environment variable in the future? Open to suggestions whether that is necessary.

Reply:
Also CC'ing @chlowell for some background here.

ADFS doesn't support oauth2 in the same way as is setup in Azure. There is no graph API etc. to go with it. For ASH the ADFS team built a custom graph service with limited capability that jury-rigged applications and service principals and some oauth2 stuff into the flow, but essentially working off of pre-authorization for applications. For example the ARM application was saved citing that any client application is authorized to request tokens for it without user consent, as if an admin had pre-consented to apps the way that can be done in Azure.

Additionally, ADFS does not translate the fake ".default" scope segment into scope values for consented permissions the way AAD does. Instead, it only supports literal scope values in a token request. In V1 token requests, there was a resource parameter that was used, alongside a scope value. In V2 flows, they combine these properties and still call it "scope" so the format becomes "resource/scope". There are also global scopes, like "openid" and "profile" and "email" which influence what claims STS include in the various tokens which aren't directly associated with any application.

In the AAD world, when you ask for a token, you provide resource/scope and AAD will look at the oauth2permissions exposed by the application described by the "resource" segment of the "scope" to then evaluate the "scope" segment of the string, with the exception of the magic ".default" scope value for SP auth to function as an alias for "all the scopes for which I have been authorized to acquire, give me the maximally powerful token" which is the design AAD went with for whatever reason.

Another exception (IIRC) is that the graph permission scopes are also "naked" so you don't need to request them with resource/scope you can use them like the global scopes.

The other relevant behavior here, is that most azure services don't use scope-based access controls, they use resource-based access controls, so only the aud claim is evaluated by them (for example, ARM doesn;t even look at the scp claim in tokens). So the scp value doesn't actaully matter for most Azure services.

Because of above, to make this Azure SDK flows work in ADFS setups, we needed ADFS to be able to resolve the target "scope" value always as a global scope value. It will still chop out the "resource" segment of a scope value to populate the aud claim, but it needs to recognize the remaining "scope" value as a global scope setup in ADFS.

Originally, we discussed with ADFS MSAL teams and agreed they would use "openid" as the scope segment for the resource/scope format so ADFS would recognize "openid" and add it as the value of the scp claim in the resulting access token.

The token "works" when calling ARM because the scope value is irrelevant, even in Azure where the token acquired would be for scope "https://management.azure.com//user_impersonation" resulting in aud claim of "https://management.azure.com/" and scp claim of "user_impersonation" (which is the default permission scope added to all applications be default and ignored for the most part).

However, for ASH stamps on recent / latest builds, we started adding "default" as a global scope built-in to ADFS so it could recognize that value on incoming calls, so code changes aren't required in the MSAL or app layers to toggle between using .default and openid based on identity provider. This results in ADFS tokens having ".default" as a literal scp claim value, instead of being translated into some permssion scope exposed by that app. This works only because the relevant azure services being called do not use scope-based auth / permissions.

Sorry for wall of text but there is some background on the behavior. For this comment specifically, please try the flow again as latest ASH ADFS environments should recognize ".default" scope value.